src/actions/vstudio/vs2005_csproj.lua

```diff
@@ -314,7 +319,61 @@
 				_x(2,'<Reference Include="%s" />', name)
 			end
 		end)
+	end
+
+	function cs2005.nuGetReferences(prj)
+		if _ACTION >= "vs2010" and prj.nuget then
+			for i = 1, #prj.nuget do
+				local package = prj.nuget[i]
+				_x(2, '<Reference Include="%s">', vstudio.nuget2010.packageId(package))
+
+				local targetFramework = vstudio.nuget2010.packageFramework(prj.solution, package)
+
+				-- We need to write HintPaths for all supported framework
+				-- versions. The last HintPath will override any previous
+				-- HintPaths (if the condition is met that is).
+
+				local function writeHintPath(frameworkVersion)
+					local assembly = vstudio.path(
+						prj,
+						p.filename(
+							prj.solution,
+							string.format(
+								"packages\\%s\\lib\\%s\\%s.dll",
+								vstudio.nuget2010.packageName(package),
+								frameworkVersion,
+								vstudio.nuget2010.packageId(package)
+							)
+						)
+					)
+
+					_x(3, '<HintPath Condition="Exists(\'%s\')">%s</HintPath>', assembly, assembly)
+				end
+
+				for k, frameworkVersion in ipairs(vstudio.frameworkVersions) do
```

Comment:
What we're doing here is building a list of .NET Framework versions up to and including the project's target version.

Maybe it would make more sense to move this logic into a function in `vstudio.cs2005` instead of in `vstudio`?

If we're going the generic route, should we also change `vstudio.frameworkVersions` to store the versions like `{ "1.0", "1.1", "2.0", "3.0", "3.5", "4.0", "4.5" }` instead of `{ "net10", "net11", "net20", "net30", "net35", "net40", "net45" }`?

This would be more consistent with the way the `dotnetframework` API works. We need to do some string transformations to convert between the two formats in either case, so it's purely a style choice.